common/src/state/mod.rs

```diff
@@ -1071,16 +1075,37 @@ impl State {
             .collect()
     }
 
-    pub fn account(&self) -> &account::Account {
-        &self.account
+    pub fn did_key(&self) -> DID {
+        self.id.clone()
     }
 
-    pub fn set_account(&mut self, account: account::Account) {
-        self.account = account;
+    pub fn set_own_identity(&mut self, identity: Identity) {
+        self.id = identity.did_key();
+        self.identities.insert(identity.did_key(), identity);
     }
 
-    pub fn set_warp_identity(&mut self, identity: warp::multipass::identity::Identity) {
-        self.account.identity.set_warp_identity(identity);
+    pub fn get_own_identity(&self) -> Identity {
+        self.identities
+            .get(&self.did_key())
+            .cloned()
+            .unwrap_or_default()
+    }
+
+    pub fn mock_own_platform(&mut self, platform: Platform) {
+        if let Some(ident) = self.identities.get_mut(&self.did_key()) {
+            ident.set_platform(platform);
+        }
+    }
+
+    // identities are updated once a minute for friends. but if someone sends you a message, they should be seen as online.
+    // this function checks if the friend is offline and if so, sets them to online. This may be incorrect, but should
+    // be corrected when the identity list is periodically updated
+    pub fn update_identity_status_hack(&mut self, id: &DID) {
```

Comment:
Not sure the intended reason for this, but using `MultiPass::identity_status` can provide the actual identity status to show if they are online, offline as well as away or busy (which can be set through `MultiPass::set_identity_status`). If they are offline (eg closed down the client and no longer connected), it wont show away or busy since it is based on their actual connection status first before the metadata.

Reply:
Actually, looking through the code, it makes a little bit more sense on the use of this function, though im wondering if there is a better way on handling this.

Reply:
yea, this function avoids an extra call to warp_runner. I think that eventually, when events are send as an identity is updated, this function would be removed.